qsimcirq/qsim_simulator.py

```diff
@@ -435,12 +435,67 @@ def compute_amplitudes_sweep_iter(
             options["s"] = self.get_seed()
             yield simulator_fn(options)
 
+    def simulate(
```

Comment:
`QSimSimulator` inherits [`cirq.SimulatesFinalState.simulate[_sweep]`](https://github.com/quantumlib/qsim/blob/5794472d347d8fec9ef89c3c29a01168cc6336c1/qsimcirq/qsim_simulator.py#L438). It does not need to implement its own version of these methods.

Reply:
it needs to allow passing the new `as_1d_state_vector` paramater